media/public/include/MediaCommon.h

```diff
@@ -332,6 +332,56 @@ enum class WebAudioPlayerState
     FAILURE        /**< The player failed to set playback state. */
 };
 
+/**
+ * @brief Additional data for decoder.
+ */
+class CodecData
```

Comment:
This file was initially design to hold some simple data types likely to be used across the system, are we happy adding a class in the public interface here?

I don't have any issue with something like this internally in Rialto server but wonder if we could keep things cleaner for the client by leaving the public interface using vector<uint8_t> and perhaps a private flag in the media segment class to indicate if the value is set.

@aczs & @skywilliamsonl what do you think?

Reply:
Yeah, in my opinion it clutters this header.

Reply:
vector<uint8_t> + flag in IMediaPipeline interface is not a good option in my opinion, because we would have to add both params to MediaSource constructor. I will change this class to a simple struct with vector and bool to keep both params in one place, user will have to remember to set both params when object is created.

Reply:
Another option is to move current class to IMediaPipeline class (like MediaSource). What do You prefer?

Reply:
As agreed on Slack, I will change it to std::shared_ptr<std::vector<uint8_t>>.